Controllers/StaffController.cs

```diff
@@ -12,34 +13,40 @@ namespace stunning_robot_HR.Controllers
 {
     public class StaffController : Controller
     {
-        private readonly stunning_robot_HRContext _context;
-
-        public StaffController(stunning_robot_HRContext context)
+        private stunning_robot_HRContext _context;
+        private  IStaffRepository staffRepository;
+        public StaffController(stunning_robot_HRContext context )
         {
             _context = context;
+            this.staffRepository = new StaffRepository(_context);
         }
-
+        
+        
         // GET: Staff
         public async Task<IActionResult> Index()
         {
-            return View(await _context.Staff.ToListAsync());
+             var staffs =  from s in staffRepository.GetStaff()
+                select s;
+             return View(staffs);
         }
-
-        // GET: Staff/Details/5
-        public async Task<IActionResult> Details(int? id)
+        
+        public ViewResult SearchStaff(string searchString)
         {
-            if (id == null)
-            {
-                return NotFound();
-            }
-
-            var staff = await _context.Staff
-                .FirstOrDefaultAsync(m => m.StaffId == id);
-            if (staff == null)
+            var staffs = from s in staffRepository.GetStaff()
+                select s;
+             
+            if (!String.IsNullOrEmpty(searchString))
             {
-                return NotFound();
+                staffs = staffs.Where(s => s.FullName.Contains(searchString));
             }
+             
+            return View(staffs);
+        }
 
+        // GET: Staff/Details/5
+        public ViewResult Details(int id)
+        {
+            Staff staff = staffRepository.GetStaffByID(id);
```

Comment:
You may want to consider using var, it's a little more flexible. That being said, developers are often times for or against it. We most often say that if you can clearly tell what type of object var represents from the right side of the expression, go with var.